src/components/Map/Map.js

```diff
@@ -3,39 +3,138 @@ import "leaflet-defaulticon-compatibility";
 import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
 import "leaflet/dist/leaflet.css";
 import { useMapEvents } from "react-leaflet";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import CustomPopup from "../Popup/Popup";
 import Tools from "../Tools/Tools";
+import * as L from "leaflet";
 
 const initialMarkers = [];
 
+const LeafIcon = L.Icon.extend({
+  options: {},
+});
+const blueIcon = new LeafIcon({
+    iconUrl:
+      "https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%E2%80%A2|FFA500&chf=a,s,ee00FFFF",
+  }),
+  greenIcon = new LeafIcon({
+    iconUrl:
+      "https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%E2%80%A2|2ecc71&chf=a,s,ee00FFFF",
+  });
+
+function getIcon(marker) {
+  if (marker.isVisited) {
+    return greenIcon;
+  }
+  return blueIcon;
+}
+
 function LocationMarker({ coords }) {
   const map = useMap();
   map.flyTo([coords.geometry.lat, coords.geometry.lng], 13);
   return null;
 }
 
+async function createMarker(marker) {
+  const response = await fetch("/api/places", {
+    method: "POST",
+    body: JSON.stringify(marker),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  const data = await response.json();
+  return data;
+}
+
+async function updateMarker(id, marker) {
+  const response = await fetch(`/api/places/${id}`, {
+    method: "PUT",
+    body: JSON.stringify(marker),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  const data = await response.json();
+  return data;
+}
+
 function Map({ searchResult, isAddingMarker }) {
   const map = useMapEvents({
     click: (e) => {
       if (!isAddingMarker) {
         return;
       }
-      console.log("clicked", e);
+
       handleAddMarker({
-        id: crypto.randomUUID(),
         lat: e.latlng.lat,
-        long: e.latlng.lng,
+        lng: e.latlng.lng,
+        isVisited: false,
+        stars: 0,
+        image: "",
+        experience: "",
       });
     },
   });
   const [markers, setMarkers] = useState(initialMarkers);
-  function handleAddMarker(marker) {
+
+  useEffect(() => {
+    async function getPlaces() {
+      const response = await fetch("/api/places");
+      const data = await response.json();
+      setMarkers(data);
+    }
+    getPlaces();
+  }, []);
+
+  async function handleAddMarker(data) {
+    const marker = await createMarker(data);
     setMarkers([...markers, marker]);
   }
 
   function handleDeleteMarker(markerId) {
-    setMarkers(markers.filter((marker) => marker.id !== markerId));
+    setMarkers(markers.filter((marker) => marker._id !== markerId));
+  }
+
+  async function handleUpdateMarker(markerId, data) {
+    try {
```

Comment:
I don't think we need the try catch on the client side. Just on the api route. 

I see you already added the try catches in the following file `src/pages/api/places/index.js` but we also need them in `src/pages/api/places/[id]/index.js`.

Lets say that the api route returns this as a response

``` js
response.status(400).json({ error: error.message });
```

then we could do something like this

``` js
const response = await updateMarker(markerId, data);

if (response.error) {
  // handle the error how you like... (maybe having a conditional error message on the page)
  return
}
const updatedMarkers = markers.map((marker) => {
  if (marker._id === markerId) {
    return { ...marker, ...data };
  }
  return marker;
});
setMarkers(updatedMarkers);
```